### DISCUSSION TITLE
Issues with sharing the same credentials between multiple app instances

### DISCUSSION THREAD

**Discussion by ipmanlk** [planetscale/discussion#467] [Q&A] [answered]
I have a database that is shared between ~100 instances (Docker containers). Right now I'm using the same credentials between all of them (with Prisma). I noticed my credentials stop working occasionally and the only possible fix I found was to create new credentials and restart my instances.  

I'm using Prisma's db push to deploy schema changes. That's pretty much it. I don't have any other configurations that could cause something like this.

Is this issue a result of using the same credentials? If so, what's the recommended way of handling something like this?

**Comment by orware** [comment]
Hello @ipmanlk,

Based on what you described, I wouldn't expect you to have an issue reusing the same set of credentials across all of those instances.

When you say that you notice your credentials stop working, is that a temporary issue that resolves after a few minutes or does it appear that the credentials don't work at all anymore?

If it's the latter, I would ask to see if you can verify if the credentials also seem to be gone from the `Settings | Passwords` area for your database too which would indicate that the credentials could have been picked up by our integration with [GitHub's Secret Scanning](https://planetscale.com/docs/concepts/connection-strings#github-secret-scanning-integration) feature, which will automatically invalidate credentials found in public repositories.

If the issue is actually more temporary and the credentials do begin working again after a few minutes, then it might be a different issue related to the load itself that may be getting sent to the database. For that, we'd probably want to dig in with further details you can share with us via ticket over at: https://support.planetscale.com/.

**Reply by ipmanlk** [reply]
Hey @orware thanks for the reply.  My password was still there in the dashboard even after this issue happened. 

I don't know if this problem is a temporary issue. I gave it ~20 minutes to see if it would resolve itself but it didn't work. I also tried to connect to the database manually when this issue occurred and the error was an invalid credentials one. 

As for password getting exposed, my project is in a private GitHub repo and it's not hard coded anywhere. It's stored in a GitHub action secret.  I confirmed there is no issue with code  itself exposing the secret either.

**Reply by orware** [reply, marked as answer]
@ipmanlk,

I appreciate those extra details! 

In that case what you shared helps to rule out the GitHub Secret Scanning integration invalidating the credentials somehow as an initial cause for this.

I'd definitely like to dig into this one further via ticket over at https://support.planetscale.com/ next time it occurs since you mentioned it seems to have happened more than once in the past. 

Even though you've resolved the most recent occurrence, I would probably still recommend submitting an initial ticket, and I can take a closer look at the database side and compile some notes to be better prepared for the next time it may occur, or share other thoughts that come out of that review process with you.
